TypeCobol/Compiler/Domain/ProgramSymbolTableBuilder.cs

```diff
@@ -0,0 +1,1727 @@
+        {
+            System.Diagnostics.Debug.Assert(dataDef.CodeElement.Type == CodeElementType.DataConditionEntry);
+            VariableSymbol sym = new VariableSymbol(dataDef.Name);
+            sym.Type = BuiltinTypes.BooleanType;
+            DecorateSymbol(dataDef, sym, parentScope);
+            if (typedef == null)
+                CurrentProgram.AddToDomain(sym);
+            else
+                typedef.Add(sym);
+            return sym;
+        }
+
+        /// <summary>
+        /// Creates a REDEFINES symbol Not typed, but with the redefined symbol resolved.
+        /// </summary>
+        /// <param name="dataDef">The DataDefinition which is a DataRedefinesEntry</param>
+        /// <param name="parentScope">The parent scope</param>
+        /// <returns>The symbol which is a RedefinesSymbol not typed if the redefined symbol is resolved, null otherwise</returns>
+        internal RedefinesSymbol CreateRedefinesSymbol(DataDefinition dataDef, Scope<VariableSymbol> parentScope)
+        {
+            System.Diagnostics.Debug.Assert(dataDef.CodeElement.Type == CodeElementType.DataRedefinesEntry);
+            //System.Diagnostics.Debug.Assert(parentScope != null);
+            DataRedefinesEntry dataRedefines = (DataRedefinesEntry) dataDef.CodeElement;
+            SymbolReference symRef = dataRedefines.RedefinesDataName;
+            if (parentScope == null)
+            {//Redefines is not supported here
+                Diagnostic d = new Diagnostic(MessageCode.SemanticTCErrorInParser,
+                    symRef.NameLiteral.Token.Column,
+                    symRef.NameLiteral.Token.EndColumn,
+                    symRef.NameLiteral.Token.Line,
+                    string.Format(TypeCobolResource.ErrRedefinesNotAllowedHere, symRef.Name));
+                Diagnostics.Add(d);
+                return null;
+            }
+
+            //Lookup the redifined symbol in the parent scope.
+            var entry = parentScope.Lookup(symRef.Name);
+            if (entry == null)
+            {//Inexisting renames
+                Diagnostic d = new Diagnostic(MessageCode.SemanticTCErrorInParser,
+                    symRef.NameLiteral.Token.Column,
+                    symRef.NameLiteral.Token.EndColumn,
+                    symRef.NameLiteral.Token.Line,
+                    string.Format(TypeCobolResource.UnknowRedefinesdSymbol, symRef.Name));
+                Diagnostics.Add(d);
+                return null;
+            }
+            //Find the Symbol which is the last in the parent scope which is not a redefines also.
+            List<VariableSymbol> previousRedefines = new List<VariableSymbol>();
+            int matchingIndex = 0;
+            for (matchingIndex = parentScope.Count - 1; matchingIndex >= 0; matchingIndex--)
+            {//Ignore all previous redefines
+                if (!parentScope[matchingIndex].HasFlag(Symbol.Flags.Redefines))
```

Comment:
No RenamesSymbol and RedefineSymbol must been as VariableSymbol thus Kinds.Variable.
So that in switch statement, we will not have to create a special case for Renames and Symbol.
The only exception is IndexSymbol.